openedx/core/djangoapps/catalog/management/commands/cache_programs.py

```diff
@@ -188,3 +202,24 @@ def process_pathways(self, site, pathways, programs):
                 logger.exception(u'Failed to process pathways for {domain}'.format(domain=site.domain))
                 failure = True
         return processed_pathways, programs, failure
+
+    def get_courses(self, programs):
+        """
+        Get all courses for the programs.
+
+        TODO: when course discovery can handle it, use that instead. That will allow us to put all course runs
+        in the cache not just the course runs in a program. Therefore, a cache miss would be different from a
+        course not in a program.
+        """
+        course_runs = {}
+        failure = False
+
+        for program_uuid, program in programs.items():
+            for course in program['courses']:
+                for course_run in course['course_runs']:
+                    course_run_key = course_run['key']
```

Comment:
If you make `course_runs = defaultdict(list)` above, then you can just use `course_runs[course_run_key].append(program_uuid)` here, rather than needing to check for membership.